admin-part/src/main/kotlin/Data.kt

```diff
@@ -234,14 +238,26 @@ data class SummaryDto(
     override fun invoke(other: Any): Any = when(other) {
         is SummaryDto -> {
             val aggCoverages = _aggCoverages + other._aggCoverages
+            val mergedTestsToRun = TestsToRun(mergeTestToRun(other))
             copy(
                 coverage = aggCoverages.average(),
                 arrow = null,
                 risks = risks + other.risks,
-                testsToRunDto = testsToRunDto, //TODO EPMDJ-2220
+                testsToRunDto = TestsToRunDto(
+                    testsToRun = mergedTestsToRun,
+                    count = mergedTestsToRun.sumSizeLists()
+                ),
                 _aggCoverages = aggCoverages
             )
         }
         else -> this
     }
+
+    private fun mergeTestToRun(other: SummaryDto): Map<String, List<String>> {
+        return (testsToRunDto.testsToRun.testTypeToNames.asSequence() + other.testsToRunDto.testsToRun.testTypeToNames.asSequence())
+            .groupBy({ it.key }, { it.value })
+            .mapValues { (_, values) ->
+                values.flatten().distinct()
```

Comment:
I think it is not well code, because I used a lot of transformation: Map -> Sequence -> Map<String, List<List<>> -> Map<String, List<String>>

I will try to find a better solution